Webservers

djodjo3 opened the pull request:
Starting to discuss about webservers in the devdocs.
Won't be able to add everything at once, but that's a start.
Let me know what you think is missing, should be reworded, etc..

matks commented on src/content/1.7/scale/webserver.md:
```diff
@@ -0,0 +1,169 @@
+
+Which is a shame as there is a lot to gain from some simple configurations parameters. And you don't need to have hundred of visitors for them to be efficient, either to reduce your server's load or speed up your visitor's browsing.
+
+As we'll see, the configuration options we'll tweak are quite easy and should allow some nice improvements.
```
On devdocs we try to have a neutral and "official" tone so we avoid shortcuts 😉
```suggestion
As we will see, the configuration options we'll tweak are quite easy and should allow some nice improvements.
```

Same for "don't need" => "do not need", and this kind of things 😄 we try to look professionnal

djodjo3 replied:
Tried to rework that in new version, let me know what you think ;)

matks commented on src/content/1.7/scale/webserver.md:
```diff
@@ -0,0 +1,169 @@
+
+# Webserver's function
+
+You may already be aware of, a webserver is a piece of software dedicated to satisfy world wide web requests.
```
Feel free to link to external resources for people who would like to discover this topic or learn more.
For example here you can link to https://en.wikipedia.org/wiki/Web_server

djodjo3 replied:
Added link.

matks commented on src/content/1.7/scale/webserver.md:
```diff
@@ -0,0 +1,169 @@
+
+## Testing your optimizations
+
+There are plenty of ways to validate the gains of those improvements but a very simple one is to go check any page speed website.
```
When I see this 😄I think "great ! let's try !" and I look for ... links to tools or services to do them. Could you add one or two speed check ?

djodjo3 replied:
I did think about it but I did not want to influence anyone using any specific tool.

djodjo3 replied:
Still, added a google search link, let me know what you think of it :p

matks commented on src/content/1.7/scale/webserver.md:
```diff
@@ -0,0 +1,169 @@
+
+### Webservers
+
+We'll show you the different ways to implement those options for both nginx and httpd, the two most popular and widespread webservers.
```
```suggestion
We'll show you the different ways to implement those options for both [nginx](https://www.nginx.com/) and [httpd](https://httpd.apache.org/docs/2.4/programs/httpd.html), the two most popular and widespread webservers.
```

Linking example 😉

djodjo3 replied:
Added links 👍

matks commented on src/content/1.7/scale/webserver.md:
```diff
@@ -0,0 +1,169 @@
+
+### Optimizations
+
+And now to the crunchy part !
```
Maybe something more neutral ? 😄 

```suggestion
Now that you know better httpd and nginx, we can see how to improve their performance.
```

djodjo3 replied:
Removed this part in new version, but crunchy ain't neutral ? 😇

matks commented on src/content/1.7/scale/webserver.md:
````diff
@@ -0,0 +1,169 @@
+LoadModule deflate_module modules/mod_deflate.so
+```
+
+Then, you can add the following lines to your VirtualHost:
````
```suggestion
Then, you can add the following lines to your VirtualHost:
```
I believe this can be done in other configuration files, right 🤔 

Remember : when you write doc, some people (junior developers or junior ops) believe What You Say Is the Truth. So if you say "put it in this file", they will think it cannot go somewhere else. You might want to say you suggest putting it in this file as this is the classical way to do it, but mention other possible areas.

djodjo3 replied:
To be clear, VirtualHost is not really a file but a configuration section (in a file, obviously ^^).
I tried to be more explicit in the new version, but paths and filenames may vary depending on versions and distribution and OS.
So, it's a bit hard to say, for sure: "add this in this file, at this path".
Maybe "supporting" some OS and version would be a solution.

toutantic commented:
nginx and apache recommandations should be on separate pages.
People reading this will look for specific instruction about the webserver they are working with and won't be interested in the other.

toutantic commented:
I believe this content could be greatly improved by following those guildelines:
https://documentation.divio.com/introduction/

The current document is a mix of explanation, how-to and reference. You should choose one type of documentation.
I believe this one should be reference or how-to but not explanations.

djodjo3 commented:
So, I've reworked this PR as follow:
* Moved the more reference/how-to parts to dedicated sections for both nginx *and* httpd
* Moved the webservers explanation part as an introduction section, so it *should* not clash between the functions

I do hope it works, but let me know what you think.

Once we have the correct structure, we should be able to add up more materials more easily.

toutantic commented:
You got it, this separation make this documentation much easier to read.
People who already know the concepts can skip directly to the how-to part, people who don't know still have the explanations !

toutantic commented:
May be replace httpd with httpd/apache2
I believe people will search more for apache than httpd ?

matks commented:
> May be replace httpd with httpd/apache2
> I believe people will search more for apache than httpd ?

Very SEO oriented 😄 

I will fix it

matks commented:
See https://github.com/PrestaShop/docs/pull/521 I did not replace all mentions but I inserted it in main titles for SEO to be optimized
